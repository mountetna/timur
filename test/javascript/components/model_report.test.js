import React from 'react';
import {Provider} from 'react-redux';
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {mockStore} from '../helpers';
import ModelReport from '../../../lib/client/jsx/components/model_report';

const models = {
  monster: {template: require('../fixtures/template_monster.json')},
  labor: {template: require('../fixtures/template_labor.json')},
  project: {template: require('../fixtures/template_project.json')}
};

describe('ModelReport', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      magma: {models}
    });
  });

  it('renders', () => {
    global.TIMUR_CONFIG = {
      magma_host: 'magma.test'
    };

    // Wrap with Provider here so store gets passed down to child components in Context
    const tree = renderer
      .create(
        <Provider store={store}>
          <ModelReport />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
