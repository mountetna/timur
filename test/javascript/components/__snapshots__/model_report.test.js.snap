// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelReport renders 1`] = `<div />`;
